src/senaite/core/content/sampletemplate.py

```diff
@@ -0,0 +1,635 @@
+from senaite.core.schema import UIDReferenceField
+from senaite.core.schema.fields import DataGridRow
+from senaite.core.z3cform.widgets.datagrid import DataGridWidgetFactory
+from senaite.core.z3cform.widgets.listing.widget import ListingWidgetFactory
+from senaite.core.z3cform.widgets.uidreference import UIDReferenceWidgetFactory
+from zope import schema
+from zope.interface import Interface
+from zope.interface import implementer
+
+
+class IServiceRecord(Interface):
+    """Record schema for selected services and partitions
+    """
+    uid = schema.TextLine(title=u"Service UID")
+    hidden = schema.Bool(title=u"Hidden")
+    part_id = schema.TextLine(title=u"Partition ID")
+
+
+class IPartitionRecord(Interface):
+    """DataGrid Row for Sample Partition Schema
+    """
+
+    # PARTITION
+    directives.widget("part_id",
+                      style=u"width:100px!important")
+    part_id = schema.TextLine(
+        title=_(
+            u"label_sampletemplate_partition_part_id",
+            default=u"Partition ID"
+        ),
+        required=False,
+        default=u"part-1")
+
+    # CONTAINER
+    directives.widget(
+        "container",
+        UIDReferenceWidgetFactory,
+        catalog=SETUP_CATALOG,
+        query={
+            "portal_type": "SampleContainer",
+            "is_active": True,
+            "sort_on": "sortable_title",
+            "sort_order": "ascending",
+        },
+        display_template="<a href='${url}'>${Title}</a>",
+        columns=get_default_columns,
+        limit=5)
+    container = UIDReferenceField(
+        title=_(
+            u"label_sampletemplate_partition_container",
+            default=u"Container"
+        ),
+        allowed_types=("SampleContainer", ),
+        multi_valued=False,
+        required=False)
+
+    # PRESERVATION
+    directives.widget(
+        "preservation",
+        UIDReferenceWidgetFactory,
+        catalog=SETUP_CATALOG,
+        query={
+            "portal_type": "SamplePreservation",
+            "is_active": True,
+            "sort_on": "sortable_title",
+            "sort_order": "ascending",
+        },
+        display_template="<a href='${url}'>${Title}</a>",
+        columns=get_default_columns,
+        limit=5)
+    preservation = UIDReferenceField(
+        title=_(
+            u"label_sampletemplate_partition_preservation",
+            default=u"Preservation"
+        ),
+        allowed_types=("SamplePreservation", ),
+        multi_valued=False,
+        required=False)
+
+    # SAMPLE TYPE
+    directives.widget(
+        "sampletype",
+        UIDReferenceWidgetFactory,
+        catalog=SETUP_CATALOG,
+        query={
+            "portal_type": "SampleType",
+            "is_active": True,
+            "sort_on": "sortable_title",
+            "sort_order": "ascending",
+        },
+        display_template="<a href='${url}'>${Title}</a>",
+        columns=get_default_columns,
+        limit=5)
```

Comment:
I would remove `limit=5`, is already the limit set by default in `QuerySelectWidget`